### PR TITLE
Adds Mocks for both Boat and ChestBoat

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Adds Mocks for Boat and ChestBoat

Progresses #825 

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	public double getMaxSpeed()
+	{
+		return this.maxSpeed;
+	}
+
+	@Override
+	@Deprecated
+	public void setMaxSpeed(double speed)
+	{
+		Preconditions.checkArgument(speed >= 0.0D, "Speed cannot be negative");
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java:94]: Looks like this silently fails in paper; if the speed is smaller than 0, the request to change it is ignored

File: src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	public double getOccupiedDeceleration()
+	{
+		return this.ocupiedDeceleration;
+	}
+
+	@Override
+	@Deprecated
+	public void setOccupiedDeceleration(double rate)
+	{
+		Preconditions.checkArgument(rate >= 0.0D, "Rate cannot be negative");
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java:109]: Same here, it silently fails in paper

File: src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java (hunk 1 above)
Comment (Insprill) [reply src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java:94]: I think it's fair to fail here. If it silently fails, the value shouldn't be less than 0, so we should alert the developer to that. Code that runs on a server doesn't *have* to run on MockBukkit, but code that runs on MockBukkit should run on a server.

Comment (Thorinwasher) [reply src/main/java/be/seeseemelk/mockbukkit/entity/BoatMock.java:94]: Good point, then I think it's fine as is

File: src/main/java/be/seeseemelk/mockbukkit/entity/ChestBoatMock.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	public long getNextRefill()
+	{
+		return this.nextRefill;
+	}
+
+	@Override
+	public long setNextRefill(long refillAt)
+	{
+		final long oldRefill = this.nextRefill;
+		this.nextRefill = refillAt;
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/ChestBoatMock.java]:
```suggestion
		this.nextRefill = (refillAt < -1) ? -1 : refillAt;
```
https://github.com/MockBukkit/MockBukkit/pull/866#discussion_r1345461088

File: src/main/java/be/seeseemelk/mockbukkit/entity/ChestBoatMock.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+				this.lootedPlayers.remove(player);
+			}
+		}
+		return hasLooted;
+	}
+
+	@Override
+	public boolean hasPendingRefill()
+	{
+		return this.nextRefill != -1;
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/entity/ChestBoatMock.java]: https://github.com/MockBukkit/MockBukkit/pull/866#discussion_r1345451630

File: src/test/java/be/seeseemelk/mockbukkit/WorldMockTest.java [hunk 5]
```diff
@@ -1278,11 +1280,14 @@ public static Stream<Arguments> getSpawnableEntities()
...
 				Arguments.of(EntityType.AREA_EFFECT_CLOUD, AreaEffectCloudMock.class),
+				Arguments.of(EntityType.BOAT, BoatMock.class),
+				Arguments.of(EntityType.CHEST_BOAT, ChestBoatMock.class),
 				Arguments.of(EntityType.FISHING_HOOK, FishHookMock.class),
 				Arguments.of(EntityType.PANDA, PandaMock.class),
 				Arguments.of(EntityType.RABBIT, RabbitMock.class),
+     		Arguments.of(EntityType.OCELOT, OcelotMock.class),
 				Arguments.of(EntityType.SLIME, SlimeMock.class),
-        		Arguments.of(EntityType.OCELOT, OcelotMock.class)
+        Arguments.of(EntityType.OCELOT, OcelotMock.class)
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/WorldMockTest.java]: This is duplicated and should be removed.

File: src/test/java/be/seeseemelk/mockbukkit/WorldMockTest.java [hunk 6]
```diff
@@ -1278,11 +1280,14 @@ public static Stream<Arguments> getSpawnableEntities()
 				Arguments.of(EntityType.MINECART, RideableMinecartMock.class),
 				Arguments.of(EntityType.MINECART_CHEST, StorageMinecartMock.class),
 				Arguments.of(EntityType.AREA_EFFECT_CLOUD, AreaEffectCloudMock.class),
+				Arguments.of(EntityType.BOAT, BoatMock.class),
+				Arguments.of(EntityType.CHEST_BOAT, ChestBoatMock.class),
 				Arguments.of(EntityType.FISHING_HOOK, FishHookMock.class),
 				Arguments.of(EntityType.PANDA, PandaMock.class),
 				Arguments.of(EntityType.RABBIT, RabbitMock.class),
+     		Arguments.of(EntityType.OCELOT, OcelotMock.class),
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/WorldMockTest.java]: Code style

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java [hunk 7]
```diff
@@ -1269,6 +1273,14 @@ else if (clazz == AreaEffectCloud.class)
...
 			return new AreaEffectCloudMock(server, UUID.randomUUID());
 		}
+		else if (clazz == Boat.class)
+		{
+			return new BoatMock(server, UUID.randomUUID());
+		}
+		else if (clazz == ChestBoat.class)
+		{
+			return new ChestBoatMock(server, UUID.randomUUID());
+    }
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]:
```suggestion
		}

**Review by Insprill — DISMISSED** [review]

**Review by Thorinwasher — APPROVED** [review]
Looks good to me.

Why haven't this been approved earlier?

**Pull request merged**